lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1445,6 +1506,14 @@ def _get_csr_secret_label(self, csr: CertificateSigningRequest) -> str:
     def _get_unit_number(self) -> str:
         return self.model.unit.name.split("/")[1]
 
+    def _validate_user_provided_private_key(self, private_key: Optional[PrivateKey]) -> bool:
```

Comment:
```suggestion
    def _user_provided_private_key(self, private_key: Optional[PrivateKey]) -> bool:
```

Reply:
I'd prefer to see something in-between here.

In my experince (not necessarily correct) `validate` implies it will preform the validation and there is no need to check anything (i.e. it will throw an error or force some other behaviour if it is invalid).

Perhaps `_is_user_provided_private_key`?

Reply:
This function has been removed in favour of raising immediately if the key passed is not valid